pkgs/tools/networking/mosh/default.nix

```diff
@@ -31,6 +31,14 @@ stdenv.mkDerivation rec {
     ./bash_completion_datadir.patch
   ];
   postPatch = ''
+    # Fix build with Xcode 12.5 toolchain/case-insensitive filesystems
+    # Backport of https://github.com/mobile-shell/mosh/commit/12199114fe4234f791ef4c306163901643b40538;
+    # remove on next upstream release.
+    patch -p0 < ${fetchpatch {
+      url = "https://raw.githubusercontent.com/macports/macports-ports/70ca3f65e622c17582fd938602d800157ed951c3/net/mosh/files/patch-version-subdir.diff";
+      sha256 = "1yyh6d07y9zbdx4fb0r56zkq9nd9knwzj22v4dfi55k4k42qxapd";
+    }}
```

Comment:
Why is this not using patches?

Reply:
I agree, this should probably be in the list of patches instead of in postPatch.

Reply:
It can't be, because `-p0` is required for only this patch. If there's a way to apply `patchFlags` to a single patch then that would be preferable, but I don't know of any.

Reply:
You're right and I also don't know.

Reply:
Can we edit the patch instead to fit into patches? I am overwriting src and patches for mosh and this breaks my other patches and causes merge conflicts. Also I don't think we should do such hackery.

Reply:
I'm doing the same thing and have the same problem.

Reply:
Sure, you can do whatever you want. I think the best change would be to allow for per-patch `patchFlags`; there's a lot of other packages using this pattern if you grep, so the inability to override flags per-patch is a persistent problem.

Reply:
No, the best change would be to adopt this patch to cleanly apply without any extra flags or is this not possible at all?

Changing how patchFlags works is going to take multiple weeks to bubble down into master and needs to be done very carefully to not break half of the package.

Reply:
There's nothing "unclean" about `-p0`. But as I said, you can do what you like.